### PR TITLE
Feature/50 users

### LINKED ISSUE #50: Add a new user into the store
## Information
When creating a new user the user should also create new logbooks for that user as to avoid the application from crashing when it is unable to fetch journals for the new user.
## Acceptance criteria
* A user is added in the store
* A weeks worth of journals are created for the given user
- Assign new id
- Assign type
- Assing email
- Assign password

## References
```ts
userObject: {
 id: number,
 name: string,
 type: string,
 email: string,
 password: string
}
```

### PR DESCRIPTION
**Pull request by te4-adrian-almetun-smeds**
## What issue did you implement or fix?
Fixed #50

## Summary
- General store improvements
- Added the ability to have multiple users in the store
- Added the ability to have journals for multiple weeks/years
- Added user classes

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## How can this implementation be improved?
- DRY it out

### CODE REVIEW

File: lib/src/store/modules/journals.ts [hunk 1]
```diff
@@ -30,19 +30,99 @@ interface CreateComment {
...
-        student: 1,
-        date: new Date(),
-        comments: [
-          { author: 1, body: "Some content", id: 1, timestamp: new Date() }
-        ]
-      }
-    ]
+    journals: {
+      "2020-48": [
+        {
```
[the remark this answers is not in the document]
Comment (te4-adrian-almetun-smeds) [reply lib/src/store/modules/journals.ts:84]: Hardcoded for now. Should be fixed in a upcoming PR.

File: lib/src/store/modules/journals.ts [hunk 2]
```diff
@@ -30,19 +30,99 @@ interface CreateComment {
...
+        },
+        {
+          id: 3,
+          student: 1,
+          date: new Date("2020-11-25"),
+          comments: [
+            { author: 1, body: "Some content", id: 1, timestamp: new Date() }
+          ]
+        },
+        {
```
[the remark this answers is not in the document]
Comment (te4-adrian-almetun-smeds) [reply lib/src/store/modules/journals.ts:108]: Hardcoded for now. Should be fixed in a upcoming PR.

File: lib/src/store/modules/journals.ts [hunk 3]
```diff
@@ -67,11 +152,42 @@ export default {
         timestamp: new Date()
       };
       state.dispatch("addComment", comment);
+    },
+    createDefaultJournals(store: any, studentId: number) {
```
[the remark this answers is not in the document]
Comment (te4-adrian-almetun-smeds) [reply lib/src/store/modules/journals.ts:156]: Hardcoded for now. Should be fixed in a upcoming PR.

**Review by te4-henrik-stahl — APPROVED** [review]
now that is a really nice code, thank you adrian man

**Pull request merged**